src/main/java/hpi/CopyDataToWorkspacePlugin.java

```diff
@@ -117,25 +131,55 @@ public boolean tearDown(AbstractBuild build, BuildListener listener)
 			}
 		};
 	}
-
-    @Override
-    public Environment setUp(Build build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-        return setUp(build, launcher, listener);
-    }
-
     @Extension
     public static class DescriptorImpl extends BuildWrapperDescriptor {
         public DescriptorImpl() {
             super(CopyDataToWorkspacePlugin.class);
         }
         
         public FormValidation doCheckFolderPath(@AncestorInPath AbstractProject project, @QueryParameter String value) throws IOException {
-            if (0 == value.length()) {
-            	return FormValidation.error("Empty path to folder");
+            if (project != null) {
+                project.checkPermission(Item.CONFIGURE);
             }
-        	return FormValidation.ok();
+            return validateFolderPath(value);
         }
 
+		public static FormValidation validateFolderPath(String value) {
+			if (value == null || value.trim().isEmpty()) {
+				return FormValidation.error("Path cannot be empty");
+			}
+
+			boolean isWindows = System.getProperty("os.name").toLowerCase().contains("windows");
+
+			if (value.contains("..") || value.contains("~")) {
+				return FormValidation.error("Path traversal characters (.., ~) are not allowed");
+			}
+
+			if (isWindows) {
+				if (value.matches("^[A-Za-z]:\\\\.*") || value.startsWith("\\")) {
+					return FormValidation.error("Absolute paths are not allowed");
+				}
+				if (value.matches(".*[<>:\"|?*].*")) {
+					return FormValidation.error("Invalid characters for Windows: <, >, :, \", |, ?, *");
+				}
+				String normalizedPath = value.replace('/', '\\');
+				if (normalizedPath.contains("\\\\")) {
+					return FormValidation.error("Consecutive path separators are not allowed");
+				}
+			} else {
+				if (value.startsWith("/")) {
+					return FormValidation.error("Absolute paths are not allowed");
+				}
+				if (value.contains("\0") || value.matches(".*[\\x00-\\x1F].*")) {

```

Comment:
The Java Character class considers a wider range of control characters with its [isISOControl()](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Character.html#isISOControl(char)) method.  I think you may want to expand your matching here or may want to consider using the Java Character class somehow.